### PR TITLE
Fix ChakraCore retail build

### PR DESCRIPTION
**Pull request by digitalinfinity**
Mostly just fixing link errors by explicitly instantiating templatized functions.
Hello world works on Linux retail builds now.

### CODE REVIEW

File: lib/Runtime/Library/JavascriptRegularExpression.h [hunk 1]
```diff
@@ -51,8 +51,15 @@ namespace Js
 
         Var GetOptions();
 
-        inline void SetPattern(UnifiedRegex::RegexPattern* pattern);
-        inline void SetSplitPattern(UnifiedRegex::RegexPattern* splitPattern);
+        void SetPattern(UnifiedRegex::RegexPattern* pattern)
```
Comment (obastemur) [inline lib/Runtime/Library/JavascriptRegularExpression.h]: IMO `inline` is fine on this new definition. Also applies to one below

**Comment by digitalinfinity** [comment]
@jianchun please review

File: lib/Runtime/Library/JavascriptRegularExpression.h (hunk 1 above)
Comment (digitalinfinity) [inline lib/Runtime/Library/JavascriptRegularExpression.h]: Yeah, i think its fine too but it's probably unnecessary. IIRC, inline is implied for member functions defined within a class definition

File: bin/ch/WScriptJsrt.cpp [hunk 2]
```diff
@@ -188,10 +188,9 @@ JsValueRef WScriptJsrt::LoadScriptHelper(JsValueRef callee, bool isConstructCall
     else
     {
         const wchar_t *fileContent;
-        char *fileName;
+        char *fileName = (char*) "script.js";
```
Comment (dilijev) [inline bin/ch/WScriptJsrt.cpp]: We're hardcoding this value?

File: lib/Parser/CharClassifier.cpp [hunk 3]
```diff
@@ -665,3 +665,11 @@ PlatformAgnostic::UnicodeText::CharacterTypeFlags Js::CharClassifier::GetCharFla
...
 }
 #endif
+
+// Explicit instantiation
+template bool Js::CharClassifier::IsIdStartFast<true>(codepoint_t) const;
+template bool Js::CharClassifier::IsIdStartFast<false>(codepoint_t) const;
+template bool Js::CharClassifier::IsIdContinueFast<true>(codepoint_t) const;
+template bool Js::CharClassifier::IsIdContinueFast<false>(codepoint_t) const;
+template bool Js::CharClassifier::IsWhiteSpaceFast<true>(codepoint_t) const;
+template bool Js::CharClassifier::IsWhiteSpaceFast<false>(codepoint_t) const;
```
Comment (dilijev) [inline lib/Parser/CharClassifier.cpp]: Don't declarations like this usually want `template <>` at the beginning (specifically the empty angle brackets)?

File: bin/ch/WScriptJsrt.cpp (hunk 2 above)
Comment (digitalinfinity) [inline bin/ch/WScriptJsrt.cpp]: Yup, this was the existing behavior- it looks like if a path is not provided, the default is script.js

File: lib/Parser/CharClassifier.cpp (hunk 3 above)
Comment (digitalinfinity) [inline lib/Parser/CharClassifier.cpp]: Template specializations do, not template instantiations

**Comment by jianchun** [comment]
LGTM

**Comment by dilijev** [comment]
LGTM

File: lib/Runtime/Library/JavascriptRegularExpression.h (hunk 1 above)
Comment (obastemur) [inline lib/Runtime/Library/JavascriptRegularExpression.h]: > IIRC, inline is implied for member functions defined within a class definition

Yes. They are actually defined within a class `JavascriptRegExp`

Comment (obastemur) [inline lib/Runtime/Library/JavascriptRegularExpression.h]: Oh. @chakrabot has merged this one without putting back `inline` ?

**Pull request merged**